validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/attestations/AttestationProductionDuty.java

```diff
@@ -136,19 +138,44 @@ private List<SafeFuture<ProductionResult<Attestation>>> produceAttestationsForCo
             CREATE_TOTAL);
     unsignedAttestationFuture.propagateTo(committee.getAttestationDataFuture());
 
+    final SignedAttestationProducer signedAttestationProducer =
+        selectSignedAttestationProducer(slot);
+
     return committee.getValidators().stream()
         .map(
             validator ->
                 signAttestationForValidatorInCommittee(
-                    slot, forkInfo, committeeIndex, validator, unsignedAttestationFuture))
+                    slot,
+                    forkInfo,
+                    committeeIndex,
+                    validator,
+                    signedAttestationProducer,
+                    unsignedAttestationFuture))
         .toList();
   }
 
+  private SignedAttestationProducer selectSignedAttestationProducer(final UInt64 slot) {
+    final SchemaDefinitions schemaDefinitions = spec.atSlot(slot).getSchemaDefinitions();
+
+    return schemaDefinitions
+        .toVersionElectra()
+        .<SignedAttestationProducer>map(
+            schemaDefinitionsElectra ->
+                (a, b, c) ->
+                    createSignedSingleAttestation(
+                        schemaDefinitionsElectra.getSingleAttestationSchema(), a, b, c))
```

Comment:
I'd rename the `a, b, c` to something more relevant: `attestationData, validator, signature`